src/blockdata/transaction.rs

```diff
@@ -840,12 +840,13 @@ impl From<EcdsaSigHashType> for u32 {
 /// This is currently returned for unrecognized sighash strings.
 #[derive(Debug, Clone)]
 pub struct SigHashTypeParseError {
-    string: String,
+    /// String representation of a sighash type which was invalid.
+    pub misspelled_ty: String,
```

Comment:
I was copying your work over in #903 and I used `unrecognized` instead of `misspelled_ty` because at another place in the code this error could be caused by a malformed hex string, 'misspelled_ty' doesn't seem to fit that use case.